rfcs/0000-top-level-assumptions.md

```diff
@@ -0,0 +1,327 @@
+# RFC: Replace "loose" options with a top-level "assumptions" object
+
+- Repo: [`babel/babel`](https://github.com/babel/babel)
+- Start Date: 2020-06-08
+- RFC PR: [babel/babel#12219](https://github.com/babel/babel/pull/12219)
+- Related Issues: <!-- if relevant -->
+- Authors: Nicolò Ribaudo
+- Champion: Nicolò Ribaudo
+- Implementors: Nicolò Ribaudo
+
+# Summary
+
+Many of our plugins have a `loose` option, which enables the compiler to make some assumptions about the code you are writing. It uses these to ignore certain edge cases and generate a smaller or faster output.
+
+The `loose` option has been around since `6to5`: at first as a top level option, and then it was split into each plugin as a catch all for anything that we wanted to be less spec compliant.
+
+However, `loose` has some problems: namely the term itself is not really descriptive at all and _should_ be named after the part of the language it affects rather than the plugin that's compiling it (see the "Motivation" section for more details). A lot of this behavior is documented with a single example, so in the future we’ll want to be specific with the assumptions themselves and maybe provide a set of tests/codesandboxes explaining the differences.
+
+This RFC proposes introducing a new *top-level* option, `"assumptions"`, which is an object containing different flags that can simplify the code that Babel generates.
+These flags have two characteristics:
+- They specify _something_ about a specific language feature, and not about a specific Babel plugin.
+- Different flags can toggle different unrelated optimizations in the same plugin, instead of toggling both of them with a single generic `"loose"` option.
+
+# Basic example
+
+```jsonc
+// babel.config.json
+{
+  "assumptions": {
+    "noDocumentAll": true,
+    "noGetterSideEffects": true,
```

Comment:
This assumption is not documented in the assumption lists. It is used in `optional-chaining` which assumes `a.b` is not an accessor.

Reply:
Thanks for catching this!